modules/aws/config.go

```diff
@@ -0,0 +1,82 @@
+package aws
+
+//Reference for mocking purposes https://github.com/aws/aws-sdk-go/blob/master/aws/config.go
+
+import (
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+)
+
+type RequestRetryer interface{}
+
+type Config struct {
+	CredentialsChainVerboseErrors *bool
+
+	Credentials *credentials.Credentials
+
+	Endpoint *string
+
+	EndpointResolver endpoints.Resolver
+
+	EnforceShouldRetryCheck *bool
+
+	Region *string
+
+	DisableSSL *bool
+
+	HTTPClient *http.Client
+
+	MaxRetries *int
+
+	Retryer RequestRetryer
+
+	DisableParamValidation *bool
+
+	DisableComputeChecksums *bool
+
+	S3ForcePathStyle *bool
+
+	S3Disable100Continue *bool
+
+	S3UseAccelerate *bool
+
+	S3DisableContentMD5Validation *bool
+
+	S3UseARNRegion *bool
+
+	EC2MetadataDisableTimeoutOverride *bool
+
+	UseDualStack *bool
+
+	SleepDelay func(time.Duration)
+
+	DisableRestProtocolURICleaning *bool
+
+	EnableEndpointDiscovery *bool
+
+	DisableEndpointHostPrefix *bool
+
+	STSRegionalEndpoint endpoints.STSRegionalEndpoint
+
+	S3UsEast1RegionalEndpoint endpoints.S3UsEast1RegionalEndpoint
+}
+
+var once sync.Once
+
+var defaultConfig *Config
```

Comment:
I assume you're doing it this way to avoid having to add extra params to all AWS methods... 

At the very least, the `NewCustomConfig` method should probably log a warning if it's called with a config value different than what was already sett...

Reply:
Hi @brikis98 the main reason is this one:

> add extra params to all AWS methods


But as you mentioned I have to take care about the concurrency. I will check it out

Reply:
Base on this comments:
Rethinked the approach

Now I'm using a custom endpoint resolver. In the test folder I include a new local file with the endpoints I like to configure locally:

1. Let me use it also in a CI pipeline
2. Avoid dealing with global configuration and possible concurrency issues, due the main goal is just be able to use it locally or in an snapshot environment


```
var LocalEndpoints = map[string]string{
	"apigateway":     "http://localhost:4567",
	"cloudformation": "http://localhost:4581",
	"cloudwatch":     "http://localhost:4582",
	"dynamodb":       "http://localhost:4569",
	"es":             "http://localhost:4578",
	"firehose":       "http://localhost:4573",
	"iam":            "http://localhost:4593",
	"kinesis":        "http://localhost:4568",
	"lambda":         "http://localhost:4574",
	"route53":        "http://localhost:4580",
	"redshift":       "http://localhost:4577",
	"s3":             "http://localhost:4572",
	"secretsmanager": "http://localhost:4584",
	"ses":            "http://localhost:4579",
	"sns":            "http://localhost:4575",
	"sqs":            "http://localhost:4576",
	"ssm":            "http://localhost:4583",
	"stepfunctions":  "http://localhost:4585",
	"sts":            "http://localhost:4592",
}

In My Test file:

	aws.SetAwsEndpointsOverrides(&LocalEndpoints)
```